pkg/sql/parser/scan.go

```diff
@@ -620,6 +620,15 @@ func (s *Scanner) scanNumber(lval *sqlSymType, ch int) {
 			return
 		}
 
+		// Strip off leading zeros from decimal literals so that
```

Comment:
Oh, trying to highlight that we only do this for non-hex literals. Updated the comment to be more clear!